player/lua/stats.lua

```diff
@@ -736,117 +743,205 @@ local function pq_eotf(x)
+end
+
+
+local function append_fps(s, prop, eprop)
+    local fps = mp.get_property_osd(prop)
+    local efps = mp.get_property_osd(eprop)
+    local single = fps ~= "" and efps ~= "" and fps == efps
+    local unit = prop == "display-fps" and " Hz" or " fps"
+    local suffix = single and "" or " (specified)"
+    local esuffix = single and "" or " (estimated)"
+    local prefix = prop == "display-fps" and "Refresh Rate:" or "Frame rate:"
+    local nl = o.nl
+    local indent = o.indent
+
+    if fps ~= "" and append(s, fps, {prefix=prefix, suffix=unit .. suffix}) then
+        prefix = ""
+        nl = ""
+        indent = ""
     end
+
+    if not single and efps ~= "" then
+        append(s, efps,
+               {prefix=prefix, suffix=unit .. esuffix, nl=nl, indent=indent})
+    end
+end
+
+
+local function add_video_out(s)
+    local vo = mp.get_property_native("current-vo")
+    if not vo then
+        return
+    end
+
+    append(s, "", {prefix=o.nl .. o.nl .. "Display:", nl="", indent=""})
+    append(s, vo, {prefix_sep="", nl="", indent=""})
+    append_property(s, "display-names", {prefix_sep="", prefix="(", suffix=")",
+                                         no_prefix_markup=true, nl="", indent=" "})
     append_property(s, "avsync", {prefix="A-V:"})
+    append_fps(s, "display-fps", "estimated-display-fps")
     if append_property(s, compat("decoder-frame-drop-count"),
                        {prefix="Dropped Frames:", suffix=" (decoder)"}) then
         append_property(s, compat("frame-drop-count"), {suffix=" (output)", nl="", indent=""})
     end
-    if append_property(s, "display-fps", {prefix="Display FPS:", suffix=" (specified)"}) then
-        append_property(s, "estimated-display-fps",
-                        {suffix=" (estimated)", nl="", indent=""})
-    else
-        append_property(s, "estimated-display-fps",
-                        {prefix="Display FPS:", suffix=" (estimated)"})
-    end
-    if append_property(s, compat("container-fps"), {prefix="FPS:", suffix=" (specified)"}) then
-        append_property(s, "estimated-vf-fps",
-                        {suffix=" (estimated)", nl="", indent=""})
-    else
-        append_property(s, "estimated-vf-fps",
-                        {prefix="FPS:", suffix=" (estimated)"})
-    end
-
     append_display_sync(s)
     append_perfdata(s, o.print_perfdata_passes)
 
-    append_resolution(s, r, "Native Resolution:", "w", "h", true)
-    if ro and (r["w"] ~= ro["dw"] or r["h"] ~= ro["dh"]) then
-        if ro["crop-w"] and (crop_noop(r["w"], r["h"], ro) or crop_equal(r, ro)) then
-            ro["crop-w"] = nil
-        end
-        append_resolution(s, ro, "Output Resolution:", "dw", "dh")
+    if mp.get_property_native("deinterlace") then
+        append_property(s, "deinterlace", {prefix="Deinterlacing:"})
     end
+
     local scale = nil
     if not mp.get_property_native("fullscreen") then
         scale = mp.get_property_native("current-window-scale")
     end
```

Comment:
also what's the idea with the scale number disappearing in fullscreen?

Reply:
I find it confusing. The `current-window-scale` is current **windowed** scale, it doesn't match the resolution in fullscreen, so I opted to hide it.

Reply:
well sure it should display a wrong value, but I expected it to display the fullscreen scale

Reply:
Something to think about, we could add similar property as `current-window-scale` for fullscreen, implementing all calc logic is not a good place in stats.lua. We can do that, but maybe in next PR, to avoid bloating this one more.

Reply:
Sure, no need to do it here or now.